subsys/lorawan/lorawan.c

```diff
@@ -0,0 +1,373 @@
+	"No network joined",		/* LORAMAC_STATUS_NO_NETWORK_JOINED */
+	"Length error",			/* LORAMAC_STATUS_LENGTH_ERROR */
+	"Region not supported",		/* LORAMAC_STATUS_REGION_NOT_SUPPORTED */
+	"Skipped APP data",		/* LORAMAC_STATUS_SKIPPED_APP_DATA */
+	"Duty-cycle restricted",	/* LORAMAC_STATUS_DUTYCYCLE_RESTRICTED */
+	"No channel found",		/* LORAMAC_STATUS_NO_CHANNEL_FOUND */
+	"No free channel found",	/* LORAMAC_STATUS_NO_FREE_CHANNEL_FOUND */
+	"Busy beacon reserved time",	/* LORAMAC_STATUS_BUSY_BEACON_RESERVED_TIME */
+	"Busy ping-slot window time",	/* LORAMAC_STATUS_BUSY_PING_SLOT_WINDOW_TIME */
+	"Busy uplink collision",	/* LORAMAC_STATUS_BUSY_UPLINK_COLLISION */
+	"Crypto error",			/* LORAMAC_STATUS_CRYPTO_ERROR */
+	"FCnt handler error",		/* LORAMAC_STATUS_FCNT_HANDLER_ERROR */
+	"MAC command error",		/* LORAMAC_STATUS_MAC_COMMAD_ERROR */
+	"ClassB error",			/* LORAMAC_STATUS_CLASS_B_ERROR */
+	"Confirm queue error",		/* LORAMAC_STATUS_CONFIRM_QUEUE_ERROR */
+	"Multicast group undefined",	/* LORAMAC_STATUS_MC_GROUP_UNDEFINED */
+	"Unknown error",		/* LORAMAC_STATUS_ERROR */
+};
+
+/* MAC event info status strings */
+const char *to_event_info_status_str[] = {
+	"OK",				/* LORAMAC_EVENT_INFO_STATUS_OK */
+	"Error",			/* LORAMAC_EVENT_INFO_STATUS_ERROR */
+	"Tx timeout",			/* LORAMAC_EVENT_INFO_STATUS_TX_TIMEOUT */
+	"Rx 1 timeout",			/* LORAMAC_EVENT_INFO_STATUS_RX1_TIMEOUT */
+	"Rx 2 timeout",			/* LORAMAC_EVENT_INFO_STATUS_RX2_TIMEOUT */
+	"Rx1 error",			/* LORAMAC_EVENT_INFO_STATUS_RX1_ERROR */
+	"Rx2 error",			/* LORAMAC_EVENT_INFO_STATUS_RX2_ERROR */
+	"Join failed",			/* LORAMAC_EVENT_INFO_STATUS_JOIN_FAIL */
+	"Downlink repeated",		/* LORAMAC_EVENT_INFO_STATUS_DOWNLINK_REPEATED */
+	"Tx DR payload size error",	/* LORAMAC_EVENT_INFO_STATUS_TX_DR_PAYLOAD_SIZE_ERROR */
+	"Downlink too many frames loss",/* LORAMAC_EVENT_INFO_STATUS_DOWNLINK_TOO_MANY_FRAMES_LOSS */
+	"Address fail",			/* LORAMAC_EVENT_INFO_STATUS_ADDRESS_FAIL */
+	"MIC fail",			/* LORAMAC_EVENT_INFO_STATUS_MIC_FAIL */
+	"Multicast fail",		/* LORAMAC_EVENT_INFO_STATUS_MULTICAST_FAIL */
+	"Beacon locked",		/* LORAMAC_EVENT_INFO_STATUS_BEACON_LOCKED */
+	"Beacon lost",			/* LORAMAC_EVENT_INFO_STATUS_BEACON_LOST */
+	"Beacon not found"		/* LORAMAC_EVENT_INFO_STATUS_BEACON_NOT_FOUND */
+};
+
+static LoRaMacPrimitives_t macPrimitives;
+static LoRaMacCallback_t macCallbacks;
+
+void OnMacProcessNotify(void)
+{
+	LoRaMacProcess();
+}
+
+static void McpsConfirm(McpsConfirm_t *mcpsConfirm)
+{
+	if (mcpsConfirm->Status != LORAMAC_EVENT_INFO_STATUS_OK) {
+		LOG_ERR("McpsRequest failed : %s",
+			log_strdup(to_event_info_status_str[mcpsConfirm->Status]));
+	} else {
+		LOG_DBG("McpsRequest success!");
+		send_status = true;
+	}
+
+	k_sem_give(&lorawan_tx_sem);
+}
+
+static void McpsIndication(McpsIndication_t *mcpsIndication)
+{
+	if (mcpsIndication->Status != LORAMAC_EVENT_INFO_STATUS_OK) {
+		LOG_ERR("%s failed : %s", __func__,
```

Comment:
Do not put `__func__` to log prints. The logger functions can be configured to add them automatically.

Reply:
Okay